ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/PackageUploader.scala

```diff
@@ -0,0 +1,130 @@
+package com.daml.ledger.participant.state.kvutils.committer
+
+import com.daml.daml_lf_dev.DamlLf.Archive
+import com.daml.ledger.participant.state.kvutils.DamlKvutils.Invalid
+import com.daml.ledger.participant.state.kvutils.committer.Committer.StepInfo
+import com.daml.lf.archive.Decode
+import com.daml.lf.data.Ref
+import com.daml.lf.engine.Engine
+import com.daml.lf.language.Ast
+import com.daml.metrics.Metrics
+
+import scala.collection.JavaConverters._
+
+/*
+ * This PackagesCommitter performs a minimal package validation before commit and
+ * upload the packages in the engine in a background process.
+ *
+ * This is design for kvutils integrations that cannot long-running submissions.
+ * When using this committer, packages on the ledger cannot be trusted and should be
+ * validated each time uploaded to the engine.
+ */
+private[committer] class PackageUploader(
+    engine: Engine,
+    metrics: Metrics
+) extends PackageCommitter(metrics) {
+
+  override protected val committerName: String = "package_upload"
+
+  metrics.daml.kvutils.committer.packageUpload.loadedPackages(() =>
```

Comment:
Not sure if this is inherited from earlier code or not, but this would only update the metric once at startup which isn't the idea. It should be done from within a step or probably preferably this is a metric that the engine should maintain and it shouldn't belong here.

Reply:
It is from the original code. 
Currently the Engine does not handle metric. If we want to do so, I will prefer wrapping it in a kv specific class. 
For now, I moved this code at the end of the preload step.